tests/test_cli.py

```diff
@@ -0,0 +1,286 @@
+        "maxplayers": "20",
+        "hostport": "9999",
+        "hostip": "192.168.56.1",
+    },
+    ["Dinnerbone", "Djinnibone", "Steve"],
+]
+
+BEDROCK_RAW_RESPONSE = [
+    "MCPE",
+    "§r§4G§r§6a§r§ey§r§2B§r§1o§r§9w§r§ds§r§4e§r§6r",
+    "422",
+    "1.18.100500",
+    "1",
+    "69",
+    "3767071975391053022",
+    "map name here",
+    "Default",
+    "1",
+    "19132",
+    "-1",
+    "3",
+]
+
+# NOTE: if updating this, be sure to change other occurrences of this help text!
+# to update, use: `COLUMNS=100000 poetry run mcstatus --help`
+EXPECTED_HELP_OUTPUT = """
+usage: mcstatus [-h] [--bedrock] address {ping,status,query,json} ...
+
+mcstatus provides an easy way to query Minecraft servers for any information they can expose. It provides three modes of access: query, status, ping and json.
+
+positional arguments:
+  address               The address of the server.
+
+options:
+  -h, --help            show this help message and exit
+  --bedrock             Specifies that 'address' is a Bedrock server (default: Java).
+
+commands:
+  Command to run, defaults to 'status'.
+
+  {ping,status,query,json}
+    ping                Ping server for latency.
+    status              Prints server status. Supported by all Minecraft servers that are version 1.7 or higher.
+    query               Prints detailed server information. Must be enabled in servers' server.properties file.
+    json                Prints server status and query in json. Supported by all Minecraft servers that are version 1.7 or higher.
+"""  # noqa: E501(line length)
+
+
+@contextlib.contextmanager
+def patch_stdout_stderr():
+    outpatch = patch("sys.stdout", new=io.StringIO())
+    errpatch = patch("sys.stderr", new=io.StringIO())
+    with outpatch as out, errpatch as err:
+        yield out, err
+
+
+@pytest.fixture
+def mock_network_requests():
+    with \
+        patch("mcstatus.server.JavaServer.lookup", return_value=JavaServer("example.com", port=25565)), \
+        patch("mcstatus.server.JavaServer.ping", return_value=0), \
+        patch("mcstatus.server.JavaServer.status", return_value=JavaStatusResponse.build(JAVA_RAW_RESPONSE)), \
+        patch("mcstatus.server.JavaServer.query", return_value=QueryResponse(*QUERY_RAW_RESPONSE)), \
+        patch("mcstatus.server.BedrockServer.lookup", return_value=BedrockServer("example.com", port=25565)), \
+        patch("mcstatus.server.BedrockServer.status", return_value=BedrockStatusResponse.build(BEDROCK_RAW_RESPONSE, latency=0)
+    ):  # fmt: skip # multiline with was added in Python 3.10
+        yield
+
+
+def normalise_help_output(s: str) -> str:
+    """
+    Normalises the output of `mcstatus --help`, to work around
+    some discrepancies between Python versions while still retaining
+    meaningful information for comparison.
+    """
+
+    elided = "[...]:"
+
+    s = s.strip()
+
+    # drop lines which end in ":". these argparse section headings vary between python versions.
```

Comment:
Switching on version is frowned upon and, for such a trivial change that's probably not explicitly documented, would require bisecting the versions.

You could maintain a manual map of replacements, I think these are very few in number. They only concern the "positional arguments", "required arguments", etc section headings.

Reply:
I see, okay then, doing that does seem fairly annoying, so let's not go there. But it might be worth explaining this a bit more in that comment for people looking at the code.

Reply:
I don't really know how to clarify it more, is what is done in c01c7ca enough?